src/rulesets-new/property-rules.ts

```diff
@@ -0,0 +1,393 @@
+import { OpenAPIV3 } from "@useoptic/openapi-utilities";
+import {
+  RequestRule,
+  ResponseBodyRule,
+  RuleError,
+  Ruleset,
+} from "@useoptic/rulesets-base";
+import { links } from "../docs";
+import { isBreakingChangeAllowed } from "./utils";
+
+const snakeCase = /^[a-z]+(?:_[a-z\d]+)*$/;
+
+const requestPropertyCasing = new RequestRule({
+  name: "request property casing",
+  rule: (requestAssertions) => {
+    requestAssertions.property.added("have snake case keys", (property) => {
+      if (!snakeCase.test(property.value.key)) {
+        throw new RuleError({
+          message: `expected ${property.value.key} to be snake case`,
+        });
+      }
+    });
+  },
+});
+
+const responsePropertyCasing = new ResponseBodyRule({
+  name: "response property casing",
+  rule: (responseAssertions) => {
+    responseAssertions.property.added("have snake case keys", (property) => {
+      if (!snakeCase.test(property.value.key)) {
+        throw new RuleError({
+          message: `expected ${property.value.key} to be snake case`,
+        });
+      }
+    });
+  },
+});
```

Comment:
You'll notice that we've split out the request and response property assertions. This is because in a lot of cases, you want to have different rules for request and response properties (e.g. adding a required request body property is a breaking change, but adding a required response body property is not).

There are obviously cases where you want these to have the same rule, but we're leaning towards verbosity, and there are ways we we can share code blocks between these functions.